File: Crio/crio_programming_interview_problems-master/HouseThief/HouseThief.java
import java.util.*;
class HouseThief {
	public static long houseThief(int n, int []arr) {
// CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
		long []dp = new long[n + 1];
		dp[0] = arr[0];
		if (n == 1)
			return dp[0];
		dp[1] = Math.max(dp[0], arr[1]);
		for (int i = 2; i < n; i++) {
			dp[i] = Math.max(dp[i - 1], (long)arr[i] + dp[i - 2]);
		}
		return dp[n - 1];
// CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

	}
	public static void main(String []args) {

		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		int []arr = new int [n];
		for ( int i = 0; i < n; i++) {
			arr[i] = sc.nextInt();
		}
		long ans = houseThief(n, arr);
		System.out.println(ans);

	}
}

File: Crio/crio_programming_interview_problems-master/HouseThief/HouseThief.py

def houseThief(n,arr):
# CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
	dp=[0 for i in range(n+1)]
	dp[0]= arr[0]
	if n==1:
		return dp[0]
	dp[1]= max( dp[0], arr[1])
	for i in range(2,n):
		dp[i]=max( dp[i-1],arr[i]+dp[i-2])
	ans=max(dp)
	return ans
# CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

def main():
	n=int(input());
	arr=list(map(int,input().split()))
	ans =houseThief(n,arr)
	print(ans)

if __name__ == '__main__':
	main()
